internal/rbd/rbd_util.go

```diff
@@ -289,6 +297,37 @@ func (rv *rbdVolume) open() (*librbd.Image, error) {
 	return image, nil
 }
 
+// allocate uses the stripe-period of the image to fully allocate (thick
+// provision) the image.
+func (rv *rbdVolume) allocate() error {
+	image, err := rv.open()
+	if err != nil {
+		return err
+	}
+	defer image.Close()
+
+	st, err := image.Stat()
+	if err != nil {
+		return err
+	}
+
+	sc, err := image.GetStripeCount()
+	if err != nil {
+		return err
+	}
+
+	// zeroBlock is the stripe-period: size of the object-size multiplied
+	// by the stripe-count
+	zeroBlock := make([]byte, sc*(1<<st.Order))
+
+	// the actual size of the image as available in the pool, can be
+	// marginally different from the requested image size
+	size := st.Size
```

Comment:
How the size can be different from the requested size here?

Reply:
the requested size gets rounded (up) on the mega bytes, so rv.Size (in MB) is different from the size of the RBD image (in bytes)